Add new UI

ofedoren opened the pull request.

ofedoren committed:
Add new UI

ofedoren committed:
Add Ansible support

ofedoren commented:
The second commit needs https://github.com/theforeman/smart_proxy_host_reports/pull/4 to work: 
![ScreenShot-1631026301862](https://user-images.githubusercontent.com/32508194/132365772-15c13f34-5763-47f6-a30b-25c6771eada8.png)

Ansible detail page is in progress.

lzap commented on app/controllers/host_reports_controller.rb:
```diff
@@ -2,4 +2,44 @@
+              id: @host_report.proxy&.id,
+              name: @host_report.proxy&.name,
+            },
+            reported_at: @host_report.reported_at.utc.to_s,
```
Is it okay to hardcode this to UTC?

ofedoren replied:
I've changed that to be the same value we store. Then it will be presented by javascript based on browser settings (lang/timezone).

lzap commented on app/models/host_report.rb:
```diff
@@ -45,8 +44,20 @@ class HostReport < ApplicationRecord
+  def status
+    @status ||= case format
+                when 'puppet'
+                  JSON.parse(body)['metrics']['resources']['values']
```
I had to add `rescue 0` in the end of this line and for ansible as well. Proabably some incorrect data in the DB?

ofedoren replied:
Yes, it will happen if you have Puppet report with a body not in `JSON.parse` compatible way stored.

lzap commented:
I can't start this up. Getting:

```
Error: A cross-origin error was thrown. React doesn't have access to the actual error object in development. See https://fb.me/react-crossorigin-error for more information.
    in StatusCell (created by BodyRow)
    in tableCell (created by BodyRow)
    in tr (created by tableRow)
    in tableRow (created by BodyRow)
    in BodyRow (created by Body)
    in tbody (created by Body)
    in Body (created by TableBody)
    in TableBody (created by Table)
    in table (created by Provider)
    in Provider (created by TablePfProvider)
    in TablePfProvider (created by Table)
    in div (created by Table)
    in Table (created by HostReportsTable)
    in HostReportsTable (created by WrappedHostReportsTable)
    in WrappedHostReportsTable (created by HostReportsIndexPage)
    in div (created by PageLayout)
    in div (created by PageLayout)
    in PageLayout (created by HostReportsIndexPage)
    in HostReportsIndexPage (created by ConnectedHostReportsIndexPage)
    in ConnectedHostReportsIndexPage (created by Context.Consumer)
    in Route
    in Switch (created by ForemanSwitcher)
    in ForemanSwitcher (created by ConnectedForemanSwitcher)
    in ConnectedForemanSwitcher (created by AppSwitcher)
    in AppSwitcher (created by ReactApp)
    in ErrorBoundary (created by ReactApp)
    in div (created by LayoutContainer)
    in LayoutContainer (created by Layout)
    in Layout (created by ConnectedLayout)
    in ConnectedLayout (created by ReactApp)
    in Router (created by ConnectedRouter)
    in ConnectedRouter (created by Context.Consumer)
    in ConnectedRouterWithContext (created by ConnectFunction)
    in ConnectFunction (created by ReactApp)
    in ApolloProvider (created by ReactApp)
    in div (created by ReactApp)
    in ReactApp (created by I18nProviderWrapper(ReactApp))
    in IntlProvider (created by I18nProviderWrapper(ReactApp))
    in I18nProviderWrapper(ReactApp) (created by StoreProvider(I18nProviderWrapper(ReactApp)))
    in Provider (created by StoreProvider(I18nProviderWrapper(ReactApp)))
    in StoreProvider(I18nProviderWrapper(ReactApp)) (created by DataProvider(StoreProvider(I18nProviderWrapper(ReactApp))))
    in DataProvider(StoreProvider(I18nProviderWrapper(ReactApp)))
```

ofedoren commented:
@lzap, updated. I've added details page for Ansible and plain formats. Ansible one will work with https://github.com/theforeman/smart_proxy_host_reports/pull/4 applied. Also, there are some fixes required in https://github.com/theforeman/foreman-ansible-modules/pull/1267.

> I can't start this up. Getting:
> 
> ```
> Error: A cross-origin error was thrown. React doesn't have access to the actual error object in development. See https://fb.me/react-crossorigin-error for more information.

I don't quite remember why it happened, but should be fixed now :)

lzap commented:
I still see:

```
Error: A cross-origin error was thrown. React doesn't have access to the actual error object in development. See https://fb.me/react-crossorigin-error for more information.,
    in StatusCell (created by BodyRow)
    in tableCell (created by BodyRow)
    in tr (created by tableRow)
    in tableRow (created by BodyRow)
    in BodyRow (created by Body)
    in tbody (created by Body)
    in Body (created by TableBody)
    in TableBody (created by Table)
    in table (created by Provider)
    in Provider (created by TablePfProvider)
    in TablePfProvider (created by Table)
    in div (created by Table)
    in Table (created by HostReportsTable)
    in HostReportsTable (created by WrappedHostReportsTable)
    in WrappedHostReportsTable (created by HostReportsIndexPage)
    in div (created by PageLayout)
    in div (created by PageLayout)
    in PageLayout (created by HostReportsIndexPage)
    in HostReportsIndexPage (created by ConnectedHostReportsIndexPage)
    in ConnectedHostReportsIndexPage (created by Context.Consumer)
    in Route
    in Switch (created by ForemanSwitcher)
    in ForemanSwitcher (created by AppSwitcher)
    in AppSwitcher (created by ReactApp)
    in ErrorBoundary (created by ReactApp)
    in div (created by LayoutContainer)
    in LayoutContainer (created by Layout)
    in Layout (created by ConnectedLayout)
    in ConnectedLayout (created by ReactApp)
    in Router (created by ConnectedRouter)
    in ConnectedRouter (created by Context.Consumer)
    in ConnectedRouterWithContext (created by ConnectFunction)
    in ConnectFunction (created by ReactApp)
    in ApolloProvider (created by ReactApp)
    in div (created by ReactApp)
    in ReactApp (created by I18nProviderWrapper(ReactApp))
    in IntlProvider (created by I18nProviderWrapper(ReactApp))
    in I18nProviderWrapper(ReactApp) (created by StoreProvider(I18nProviderWrapper(ReactApp)))
    in Provider (created by StoreProvider(I18nProviderWrapper(ReactApp)))
    in StoreProvider(I18nProviderWrapper(ReactApp)) (created by DataProvider(StoreProvider(I18nProviderWrapper(ReactApp))))
    in DataProvider(StoreProvider(I18nProviderWrapper(ReactApp)))
```

There is a JS test failing FYI.

lzap commented:
Merged in #7
